.github/workflows/pr-vrt-comment.yml

```diff
@@ -34,134 +31,104 @@ jobs:
       # downloaded artifacts will contain screenshots from affected project including 'screenshots-report.json' which contains proper image mappings for affected project
       # - see @{link file://./../scripts/prepare-vr-screenshots-for-upload.js#43}
       # - see @{link file://./pr-vrt.yml#56}
-      - uses: actions/download-artifact@v4
+
+      - name: Check for vrscreenshot artifact
+        id: check_artifact
+        uses: actions/github-script@v7
         with:
-          name: vrscreenshot
-          path: ./screenshots
-          run-id: ${{ github.event.workflow_run.id }}
-          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const artifacts = await github.rest.actions.listWorkflowRunArtifacts({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              run_id: context.payload.workflow_run.id
```

Comment:
AFAIR this won't work ? - context doesn't provide this information - that's why we generate PR id artifact on pull_request trigger to be consumed within privileged pipeline

Reply:
This is available. Check https://docs.github.com/en/webhooks/webhook-events-and-payloads#workflow_run